### PR TITLE
Reinstate dump compress

### LINKED ISSUE #1067: Reinstate zlib_compression option
There's been a number of reports of breakage from new latest tskit release in which the removal of the ``zlib_compression`` option has broken people's code. (.e.g. https://github.com/tskit-dev/msprime/issues/1327)

We should revert the change, and push out a bugfix release ASAP.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Fixes #1067 and does the release prep. Should be good to cut a release tomorrow then @benjeffery?

Also, I think this answers the question about whether people are using the msprime ``msp`` CLI!

### CODE REVIEW

File: python/CHANGELOG.rst [hunk 1]
```diff
@@ -1,13 +1,14 @@
 --------------------
-[0.X.X] - 2021-XX-XX
+[0.3.4] - 2021-12-02
```
Comment (benjeffery) [inline python/CHANGELOG.rst:2]: My guess of the year was wrong!
```suggestion
[0.3.4] - 2020-12-02
```

File: python/tests/test_highlevel.py [hunk 2]
```diff
@@ -1288,6 +1288,17 @@ def test_dump_load_errors(self):
...
+    def test_zlib_compression_warning(self, ts_fixture):
+        with tempfile.TemporaryDirectory() as tempdir:
+            temp_file = pathlib.Path(tempdir) / "tmp.trees"
+            with warnings.catch_warnings(record=True) as w:
+                ts_fixture.dump(temp_file, zlib_compression=True)
+                assert len(w) == 1
+                assert issubclass(w[0].category, RuntimeWarning)
+            with warnings.catch_warnings(record=True) as w:
+                ts_fixture.dump(temp_file, zlib_compression=False)
+                assert len(w) == 0
```
Comment (benjeffery) [inline python/tests/test_highlevel.py:1300]:
```suggestion
    def test_zlib_compression_warning(self, ts_fixture, temp_path):
        temp_file = temp_path / "tmp.trees"
        with warnings.catch_warnings(record=True) as w:
            ts_fixture.dump(temp_file, zlib_compression=True)
            assert len(w) == 1
            assert issubclass(w[0].category, RuntimeWarning)
        with warnings.catch_warnings(record=True) as w:
            ts_fixture.dump(temp_file, zlib_compression=False)
            assert len(w) == 0
```

**Review by benjeffery — APPROVED** [review]
LGTM - couple of nits!

**Comment by jeromekelleher** [comment]
Haha, thanks @benjeffery - maybe I shouldn't be doing release prep during the graveyard shift!

I'm happy to merge with the extra commit, or can squash, as you prefer.

**Comment by benjeffery** [comment]
@jeromekelleher sorry, my memory failed me on the fixture name. Should be tmp_path.

**Comment by jeromekelleher** [comment]
Done and squashed; ta.

**Comment by benjeffery** [comment]
Great I'll merge this in and release in the morning with a (hopefully) fresh head.

**Pull request merged**